src/lib/homestore_backend/hs_shard_manager.cpp

```diff
@@ -102,8 +102,44 @@ ShardManager::AsyncResult< ShardInfo > HSHomeObject::_create_shard(pg_id_t pg_ow
     });
 }
 
-ShardManager::Result< ShardInfo > HSHomeObject::_seal_shard(shard_id_t id) {
-    return folly::makeUnexpected(ShardError::UNKNOWN_SHARD);
+ShardManager::AsyncResult< ShardInfo > HSHomeObject::_seal_shard(ShardInfo const& info) {
+    auto& pg_id = info.placement_group;
+    auto& shard_id = info.id;
+    ShardInfo shard_info = info;
+
+    shared< homestore::ReplDev > repl_dev;
+    {
+        std::shared_lock lock_guard(_pg_lock);
+        auto iter = _pg_map.find(pg_id);
+        RELEASE_ASSERT(iter != _pg_map.end(), "PG not found");
+        repl_dev = static_cast< HS_PG* >(iter->second.get())->repl_dev_;
+        RELEASE_ASSERT(repl_dev != nullptr, "Repl dev null");
+    }
+
+    shard_info.state = ShardInfo::State::SEALED;
+    auto seal_shard_message = serialize_shard_info(shard_info);
+    const auto msg_size = sisl::round_up(seal_shard_message.size(), repl_dev->get_blk_size());
+    auto req = repl_result_ctx< ShardManager::Result< ShardInfo > >::make(msg_size, 512 /*alignment*/);
+    auto buf_ptr = req->hdr_buf_.bytes;
+    std::memset(buf_ptr, 0, msg_size);
+    std::memcpy(buf_ptr, seal_shard_message.c_str(), seal_shard_message.size());
+
+    req->header_.msg_type = ReplicationMessageType::SEAL_SHARD_MSG;
+    req->header_.pg_id = pg_id;
+    req->header_.shard_id = shard_id;
+    req->header_.payload_size = msg_size;
+    req->header_.payload_crc = crc32_ieee(init_crc32, buf_ptr, msg_size);
+    req->header_.seal();
+    sisl::blob header;
+    header.bytes = r_cast< uint8_t* >(&req->header_);
+    header.size = sizeof(req->header_);
+    sisl::sg_list value;
+    value.size = msg_size;
+    value.iovs.push_back(iovec(buf_ptr, msg_size));
+
+    // replicate this seal shard message to PG members;
+    repl_dev->async_alloc_write(header, sisl::blob{}, value, req);
```

Comment:
We also need to update the shard's metablk from active to sealed state, right?

Reply:
It was already there in hs_shard_manager.cpp on commit